ovirt_imageio/client/_ui.py

```diff
@@ -101,24 +109,41 @@ def close(self):
 
     def _draw(self, last=False):
         line = self._format_line()
+        if self._format and not last:
+            line += ','
 
         # Using "\r" moves the cursor to the first column, so the next progress
-        # will overwrite this one. If this is the last progress, we use "\n" to
-        # move to the next line. Otherwise, the next shell prompt will include
-        # part of the old progress.
-        end = "\n" if last else "\r"
+        # will overwrite this one. If this is the last progress or there is a
+        # non-default format set, we use "\n" to move to the next line.
+        # Otherwise, the next shell prompt will include part of the old
+        # progress.
+        end = "\n" if last or self._format else "\r"
```

Comment:
Don't make this logic more complicated - instead use a writer object to format
and write the lines, so this logic is only in the text writer.